### PR TITLE
Python3 write safe (Codacy dislikes pytest)

### PR DESCRIPTION
**Pull request by wxtim**
Spun out _write_safe method into a new unicode_utils module

### CODE REVIEW

File: lib/python/rose/unicode_utils.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+            Message to be written - type uncertain.
+        handle (file handle):
+            Potentially Obscure file or file-like object. May include a variety of
+            buffer types.
+    """
+    if isinstance(handle, io.BufferedWriter) and isinstance(msg, str):
+        handle.write(msg.encode(ENCODING))
+    elif isinstance(handle, io.StringIO) and isinstance(msg, bytes):
+        handle.write(msg.decode())
+        print('N')
```
Comment (matthewrmshin) [inline lib/python/rose/unicode_utils.py]: Debug left over?

File: lib/python/rose/unicode_utils.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    elif isinstance(handle, io.StringIO) and isinstance(msg, str):
+        handle.write(msg)
+    else:
+        raise TypeError(f'Unsupported types:'
+          f'\t{msg} has type {type(msg)}'
+          f'\t{handle} has type {type(handle)}')
+
+
+if __name__ == "__main__":
+    pass
```
Comment (matthewrmshin) [inline lib/python/rose/unicode_utils.py]: No point to have this block?

File: lib/python/rose/unicode_utils.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        handle.write(msg.decode())
+        print('N')
+    elif isinstance(handle, io.BufferedWriter) and isinstance(msg, bytes):
+        handle.write(msg)
+    elif isinstance(handle, io.StringIO) and isinstance(msg, str):
+        handle.write(msg)
+    else:
+        raise TypeError(f'Unsupported types:'
+          f'\t{msg} has type {type(msg)}'
+          f'\t{handle} has type {type(handle)}')
```
Comment (matthewrmshin) [inline lib/python/rose/unicode_utils.py]: Not a good day for polymorphism, but I don't really have any good alternate suggestion.

File: lib/python/rose/unicode_utils.py (hunk 1 above)
Comment (wxtim) [reply lib/python/rose/unicode_utils.py]: Yes.

File: lib/python/rose/unicode_utils.py (hunk 2 above)
Comment (wxtim) [reply lib/python/rose/unicode_utils.py]: None at all.

**Comment by wxtim** [comment]
#### RE: `assert==value` pytests fail in codacy
https://github.com/fossasia/query-server/issues/332 seems informative, if not terribly helpful (all the links to codacy are broken).

File: .bandit [hunk 4]
```diff
@@ -0,0 +1 @@
+skips: ['B101']
```
Comment (oliver-sanders) [inline .bandit]: I'm happy with this as we have little to gain from running using Python's [so-called] optimisation.

**Pull request merged**